### PR TITLE
Updated audit logs and id changes for roles, request action

### PR DESCRIPTION
**Pull request by neerk28**

### CODE REVIEW

File: functions/api/pendingTransactions.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    const allPendingTransactions = []
+    let pendingTransactions = db.collection(constants.BRANCHES).doc(branchId).collection(constants.PENDING_TRANSACTIONS);
+    let snapshot = await pendingTransactions.get()
+    snapshot.forEach(transaction => {
+        const trxData = transaction.data()
+        trxData[constants.DATE] = trxData[constants.DATE].toDate()
+        allPendingTransactions.push({ id: transaction.id, ...trxData })
+    })
+    const response = {
+        pendingTransactions: allPendingTransactions,
```
Comment (shrikantdange1208) [inline functions/api/pendingTransactions.js]: Should it be pendingRequests?
Also for constant PENDING_REQUEST

File: functions/api/roles.js [hunk 2]
```diff
@@ -97,21 +96,24 @@ router.put('/', isAdmin, async (req, res, next) => {
...
+    delete newData[constants.ID]
+    delete newData[constants.CREATED_DATE]
+    newData[constants.LAST_UPDATED_DATE] = new Date()
+    await usersRef.set(newData, { merge: true })
+    newData[constants.CREATED_DATE] = oldData[constants.CREATED_DATE]
 
     // Fire and forget audit log
-    const eventMessage = `User ${req.user.firstName} updated role ${id.toLocaleLowerCase()}`
+    const eventMessage = `User ${req.user.firstName} updated role ${oldData[constants.NAME]}`
     audit.logEvent(eventMessage, req)
```
Comment (shrikantdange1208) [inline functions/api/roles.js]: In case of update, you will have to send oldData and newData in logEvent

File: functions/api/roles.js [hunk 3]
```diff
@@ -145,19 +148,19 @@ function validateInput(body, type) {
 
         case constants.CREATE:
             schema = joi.object().keys({
-                label: joi.string().regex(/^[a-zA-Z]{5,10}$/).required(),
+                name: joi.string().regex(/^[a-zA-Z]{5,10}$/).required(),
                 description: joi.string().regex(/^[a-z A-Z]{5,40}$/).required(),
                 permissions: joi.array().items(joi.string().required()).required(),
                 isActive: joi.bool()
```
Comment (shrikantdange1208) [inline functions/api/roles.js]: Comment on old code. isActive should be set to required.

**Review by shrikantdange1208 — CHANGES_REQUESTED** [review]

File: functions/api/roles.js (hunk 2 above)
Comment (neerk28) [reply functions/api/roles.js]: good catch :)

File: functions/api/pendingTransactions.js (hunk 1 above)
Comment (neerk28) [reply functions/api/pendingTransactions.js]: yes, will change

File: functions/api/roles.js (hunk 3 above)
Comment (neerk28) [reply functions/api/roles.js]: yep

**Review by shrikantdange1208 — APPROVED** [review]

**Pull request merged**